101-azure-machine-learning-service-workspace/azuredeploy.json

```diff
@@ -0,0 +1,123 @@
+        },
+        "keyVault": {
+            "name": "[concat('kv',uniqueString(resourceGroup().id))]",
+            "tenantId": "[subscription().tenantId]"
+        },
+        "applicationInsightsName": "[concat('ai',uniqueString(resourceGroup().id))]",
+        "containerRegistryName": "[concat('cr',uniqueString(resourceGroup().id))]"
+    },
+    "resources": [
+        {
+            "name": "[variables('storageAccount').name]",
+            "type": "Microsoft.Storage/storageAccounts",
+            "location": "[parameters('location')]",
+            "apiVersion": "2018-07-01",
+            "sku": {
+                "name": "[variables('storageAccount').type]"
+            },
+            "kind": "StorageV2",
+            "properties": {
+                "encryption": {
+                    "services": {
+                        "blob": {
+                            "enabled": true
+                        },
+                        "file": {
+                            "enabled": true
+                        }
+                    },
+                    "keySource": "Microsoft.Storage"
+                },
+                "supportHttpsTrafficOnly": true
+            }
+        },
+        {
+            "name": "[variables('keyVault').name]",
+            "type": "Microsoft.KeyVault/vaults",
+            "apiVersion": "2018-02-14",
+            "location": "[parameters('location')]",
+            "properties": {
+                "tenantId": "[variables('keyVault').tenantId]",
+                "sku": {
+                    "name": "standard",
+                    "family": "A"
+                },
+                "accessPolicies": []
+            }
+        },
+        {
+            "name": "[variables('applicationInsightsName')]",
+            "type": "Microsoft.Insights/components",
+            "apiVersion": "2015-05-01",
+            "location": "[parameters('location')]",
+            "kind": "web",
+            "properties": {
+                "Application_Type": "web"
+            }
+        },
+        {
+            "name": "[variables('containerRegistryName')]",
+            "type": "Microsoft.ContainerRegistry/registries",
+            "apiVersion": "2017-10-01",
+            "location": "[parameters('location')]",
+            "sku": {
+                "name": "Standard"
+            },
+            "properties": {
+                "adminUserEnabled": true
+            }
+        },
+        {
+            "name": "[parameters('workspaceName')]",
+            "type": "Microsoft.MachineLearningServices/workspaces",
+            "apiVersion": "2018-11-19",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[variables('storageAccount').name]",
+                "[variables('keyVault').name]",
+                "[variables('applicationInsightsName')]",
+                "[variables('containerRegistryName')]"
+            ],
+            "identity": {
```

Comment:
how is this used?

Reply:
When the workspace is created, a managed system identity is created and associated with it. When the workspace is deleted, the managed system identity is also removed. It's required by the workspace.